secboot/secboot_tpm.go

```diff
@@ -393,23 +426,37 @@ func SealKeys(keys []SealKeyRequest, params *SealKeysParams) ([]byte, error) {
 		if err := key.BootstrappedContainer.AddKey(key.SlotName, unlockKey); err != nil {
 			return nil, err
 		}
-		writer := sb.NewFileKeyDataWriter(key.KeyFile)
-		if err := protectedKey.WriteAtomic(writer); err != nil {
+
+		var keyWriter sb.KeyDataWriter
+		if key.KeyFile != "" {
+			keyWriter = sb.NewFileKeyDataWriter(key.KeyFile)
+		} else {
+			tokenWriter, err := key.BootstrappedContainer.GetTokenWriter(key.SlotName)
+			if err != nil {
+				return nil, err
+			}
+			keyWriter = tokenWriter
+		}
```

Comment:
Tiny nitpick, This could be shared with `SealKeysWithFDESetupHook`

Reply:
Yes, it actually made more sense as a method to SealKeyRequest and ended up with 5 lines.